### ISSUE TITLE
FT.SEARCH takes over 100ms for simple queries (WSL)

### ISSUE THREAD

**Issue by nrhbpl** [RediSearch/RediSearch#3116]
Hi,

I’m trying to understand what could explain high latency (>100ms) in basic FT.SEARCH commands.

**TLDR;**

- All tests were performed on local machine (running both Redis server and client)
- FT.SEARCH takes over 100ms for a very simple query
- Running the **same** query with redis-benchmark shows it’s executed in less than 0.5 millisecond (maximum execution time on 50 000 calls = 2ms)
- The slow FT.SEARCH result was obtained from a dotnet console app (using NRediSearch package)
- I confirmed that the latency **is outside of the dotnet console app** (listening on loopback interface with Wireshark shows more than 100ms between the network request sent to Redis and the response containing the result)

My setup:
- WSL2 (Ubuntu 20.04)
- Up to date Windows 10 laptop with plenty of available resources
- Created 1M HSET (with only 1 field named “description” where I store about 1KB of text for each element)
- The command to create the index: `FT.CREATE "idx:string" ON hash PREFIX 1 "string:" SCHEMA description TEXT`
- The command to run the search: `FT.SEARCH "idx:string" "temp*" LIMIT 0 10`
- I tried to run those tests with my antivirus disabled, same result
- Running `redis-benchmark -n 50000 -P 1 script load "redis.call('FT.SEARCH idx:string temp*')"` shows that it takes less than 0.5ms to execute the search request (the slowest of the 50000 calls is just over 2 ms).
 
At first, I investigated in the dotnet application and the library I used as I suspected that’s where the latency was coming from (given the redis-benchmark results). After some digging everything seemed fine, and it really seemed like there was some latency between my app and my Redis instance. The redis-benchmark commands were executed within WSL but my console app was running on Windows, so I installed dotnet on WSL and ran my application from there but the result was the same.

I ran again my application from Windows with Wireshark listening on local loopback to realize that there was in fact a latency somewhere between the app and Redis (not within the app). We can see the network package for the FT.SEARCH command and, about 100ms later, we can see the response passing on the “wire”. The capture I made when executing MGET commands show the packet exchange happens well within 1ms.

I’d appreciate any input that could help me understand the source of the latency. I can provide redis-benchmark results and Wireshark pcap files if needed.

Thanks to all of you for providing this project to the open source community!

**Comment by filipecosta90** [comment]
hi there @nrhbpl can you provide more details on how you're running redisearch and which version. 
Furthermore you can simplify the benchmark command using solely:
```
redis-benchmark -n 50000 FT.SEARCH idx:string temp*
```
As a second step can you provide the output of:
1)
```
redis-cli INFO
```
2)
```
redis-cli INFO COMMANDSTATS
```
and
3)
```
redis-cli FT.PROFILE idx:string SEARCH QUERY "temp*"
```
. 
That should be able to give us further insights on your issue.

**Comment by nrhbpl** [comment]
Thanks a lot for the prompt response Filipe,

Using your command for redis-benchmark I've got massively different results, it's even slower than the tests I were running from my dotnet app and I had to lower the number of requests to limit the waiting time. I've added below the result of every command you mentionned in your reply. The profile result seems to indicate that the SORT is the source of the slow commands...

`redis-benchmark -n 50000 FT.SEARCH idx:string temp*`
======_ FT.SEARCH idx:string temp* ======
  500 requests completed in 30.45 seconds
  50 parallel clients
  47 bytes payload
  keep alive: 1
  host configuration "save": 3600 1 300 100 60 10000
  host configuration "appendonly": no
  multi-thread: no

Latency by percentile distribution:
0.000% <= 98.239 milliseconds (cumulative count 1)
50.000% <= 3000.319 milliseconds (cumulative count 500)
100.000% <= 3000.319 milliseconds (cumulative count 500)

Cumulative distribution of latencies:
0.000% <= 0.103 milliseconds (cumulative count 0)
0.200% <= 99.135 milliseconds (cumulative count 1)
0.600% <= 178.175 milliseconds (cumulative count 3)
1.400% <= 1343.487 milliseconds (cumulative count 7)
3.400% <= 1344.511 milliseconds (cumulative count 17)
3.600% <= 1402.879 milliseconds (cumulative count 18)
4.000% <= 1403.903 milliseconds (cumulative count 20)
5.800% <= 1409.023 milliseconds (cumulative count 29)
6.600% <= 1507.327 milliseconds (cumulative count 33)
7.400% <= 1738.751 milliseconds (cumulative count 37)
8.200% <= 1739.775 milliseconds (cumulative count 41)
8.800% <= 1812.479 milliseconds (cumulative count 44)
10.000% <= 1833.983 milliseconds (cumulative count 50)
10.400% <= 2047.999 milliseconds (cumulative count 52)
12.000% <= 2049.023 milliseconds (cumulative count 60)
13.600% <= 2067.455 milliseconds (cumulative count 68)
17.400% <= 2433.023 milliseconds (cumulative count 87)
25.000% <= 2445.311 milliseconds (cumulative count 125)
30.600% <= 2719.743 milliseconds (cumulative count 153)
31.000% <= 2838.527 milliseconds (cumulative count 155)
36.400% <= 2926.591 milliseconds (cumulative count 182)
38.200% <= 2953.215 milliseconds (cumulative count 191)
46.200% <= 2955.263 milliseconds (cumulative count 231)
100.000% <= 3000.319 milliseconds (cumulative count 500)
100.000% <= 3000.319 milliseconds (cumulative count 500)

Summary:
  throughput summary: 16.42 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
     2721.949    98.176  3000.319  3000.319  3000.319  3000.319


`redis-cli INFO`
# Server
redis_version:7.0.4
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:c7d71d4b63066c
redis_mode:standalone
os:Linux 4.4.0-19041-Microsoft x86_64
arch_bits:64
monotonic_clock:POSIX clock_gettime
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:9.4.0
process_id:108
process_supervised:no
run_id:42b04d3a26d461d06429118406daa00209a6c35e
tcp_port:6379
server_time_usec:1664288942467671
uptime_in_seconds:314841
uptime_in_days:3
hz:10
configured_hz:10
lru_clock:3344558
executable:/usr/bin/redis-server
config_file:/etc/redis/redis.conf
io_threads_active:0

# Clients
connected_clients:4
cluster_connections:0
maxclients:10000
client_recent_max_input_buffer:8
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:2066466960
used_memory_human:1.92G
used_memory_rss:2155409408
used_memory_rss_human:2.01G
used_memory_peak:4930365288
used_memory_peak_human:4.59G
used_memory_peak_perc:41.91%
used_memory_overhead:49382744
used_memory_startup:987616
used_memory_dataset:2017084216
used_memory_dataset_perc:97.66%
allocator_allocated:2067735576
allocator_active:2071506944
allocator_resident:2225012736
total_system_memory:25475211264
total_system_memory_human:23.73G
used_memory_lua:34816
used_memory_vm_eval:34816
used_memory_lua_human:34.00K
used_memory_scripts_eval:936
number_of_cached_scripts:6
number_of_functions:0
number_of_libraries:0
used_memory_vm_functions:32768
used_memory_vm_total:67584
used_memory_vm_total_human:66.00K
used_memory_functions:184
used_memory_scripts:1120
used_memory_scripts_human:1.09K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.00
allocator_frag_bytes:3771368
allocator_rss_ratio:1.07
allocator_rss_bytes:153505792
rss_overhead_ratio:0.97
rss_overhead_bytes:-69603328
mem_fragmentation_ratio:1.04
mem_fragmentation_bytes:88982360
mem_not_counted_for_evict:0
mem_replication_backlog:0
mem_total_replication_buffers:0
mem_clients_slaves:0
mem_clients_normal:5400
mem_cluster_links:0
mem_aof_buffer:0
mem_allocator:jemalloc-5.2.1
active_defrag_running:0
lazyfree_pending_objects:0
lazyfreed_objects:0

# Persistence
loading:0
async_loading:0
current_cow_peak:0
current_cow_size:0
current_cow_size_age:0
current_fork_perc:0.00
current_save_keys_processed:0
current_save_keys_total:0
rdb_changes_since_last_save:1000002
rdb_bgsave_in_progress:0
rdb_last_save_time:1664287386
rdb_last_bgsave_status:err
rdb_last_bgsave_time_sec:0
rdb_current_bgsave_time_sec:-1
rdb_saves:44331
rdb_last_cow_size:0
rdb_last_load_keys_expired:0
rdb_last_load_keys_loaded:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_rewrites:0
aof_rewrites_consecutive_failures:0
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0

# Stats
total_connections_received:3070
total_commands_processed:18437408
instantaneous_ops_per_sec:0
total_net_input_bytes:9050025017
total_net_output_bytes:1907314778
total_net_repl_input_bytes:0
total_net_repl_output_bytes:0
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
instantaneous_input_repl_kbps:0.00
instantaneous_output_repl_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:3548
evicted_keys:0
evicted_clients:0
total_eviction_exceeded_time:0
current_eviction_exceeded_time:0
keyspace_hits:16602707
keyspace_misses:95
pubsub_channels:0
pubsub_patterns:0
pubsubshard_channels:0
latest_fork_usec:123686
total_forks:44355
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
total_active_defrag_time:0
current_active_defrag_time:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0
total_error_replies:400177
dump_payload_sanitizations:0
total_reads_processed:3063991
total_writes_processed:3095479
io_threaded_reads_processed:0
io_threaded_writes_processed:0
reply_buffer_shrinks:1591
reply_buffer_expands:708

# Replication
role:master
connected_slaves:0
master_failover_state:no-failover
master_replid:6d0bd31b52038142b7479813bfbf01315949b8e1
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:11115.218750
used_cpu_user:1266.343750
used_cpu_sys_children:8495.390625
used_cpu_user_children:1422.500000
used_cpu_sys_main_thread:11082.796875
used_cpu_user_main_thread:1123.671875

# Modules
module:name=search,ver=999999,api=1,filters=0,usedby=[],using=[],options=[]

# Errorstats
errorstat_ERR:count=167
errorstat_Index:count=1
errorstat_LIMIT:count=1
errorstat_MISCONF:count=400000
errorstat_Unknown:count=8

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=1000000,expires=0,avg_ttl=0


`redis-cli INFO COMMANDSTATS`
# Commandstats
cmdstat_module|list:calls=4,usec=27,usec_per_call=6.75,rejected_calls=0,failed_calls=0
cmdstat_config|set:calls=1,usec=36,usec_per_call=36.00,rejected_calls=0,failed_calls=0
cmdstat_config|get:calls=265,usec=3433,usec_per_call=12.95,rejected_calls=0,failed_calls=0
cmdstat_ping:calls=200055,usec=52227,usec_per_call=0.26,rejected_calls=0,failed_calls=0
cmdstat_mset:calls=100000,usec=637324,usec_per_call=6.37,rejected_calls=0,failed_calls=0
cmdstat_echo:calls=156,usec=326,usec_per_call=2.09,rejected_calls=0,failed_calls=0
cmdstat_info:calls=820,usec=752901,usec_per_call=918.17,rejected_calls=0,failed_calls=0
cmdstat_sadd:calls=100000,usec=36771,usec_per_call=0.37,rejected_calls=0,failed_calls=0
cmdstat_lpush:calls=200000,usec=97863,usec_per_call=0.49,rejected_calls=0,failed_calls=0
cmdstat_llen:calls=2,usec=8,usec_per_call=4.00,rejected_calls=0,failed_calls=0
cmdstat_client|setname:calls=170,usec=1630,usec_per_call=9.59,rejected_calls=0,failed_calls=0
cmdstat_FT.DROP:calls=11,usec=14881550,usec_per_call=1352868.12,rejected_calls=0,failed_calls=3
cmdstat_memory|usage:calls=17544,usec=80386,usec_per_call=4.58,rejected_calls=0,failed_calls=0
cmdstat_hgetall:calls=68,usec=1402,usec_per_call=20.62,rejected_calls=0,failed_calls=0
cmdstat_type:calls=17544,usec=19803,usec_per_call=1.13,rejected_calls=0,failed_calls=0
cmdstat_FT.MGET:calls=11,usec=2680,usec_per_call=243.64,rejected_calls=0,failed_calls=0
cmdstat_FT.INFO:calls=2,usec=2084,usec_per_call=1042.00,rejected_calls=0,failed_calls=2
cmdstat_lpop:calls=100000,usec=127948,usec_per_call=1.28,rejected_calls=0,failed_calls=0
cmdstat_FT.CREATE:calls=11,usec=8877,usec_per_call=807.00,rejected_calls=0,failed_calls=5
cmdstat_subscribe:calls=76,usec=1187,usec_per_call=15.62,rejected_calls=0,failed_calls=0
cmdstat_incr:calls=100000,usec=91886,usec_per_call=0.92,rejected_calls=0,failed_calls=0
cmdstat_FT._LIST:calls=2,usec=42,usec_per_call=21.00,rejected_calls=0,failed_calls=0
cmdstat_cluster|nodes:calls=80,usec=244,usec_per_call=3.05,rejected_calls=0,failed_calls=80
cmdstat_lindex:calls=1,usec=10,usec_per_call=10.00,rejected_calls=2,failed_calls=0
cmdstat_spop:calls=100000,usec=31701,usec_per_call=0.32,rejected_calls=0,failed_calls=0
cmdstat_exists:calls=70,usec=478,usec_per_call=6.83,rejected_calls=0,failed_calls=0
cmdstat_strlen:calls=3,usec=24,usec_per_call=8.00,rejected_calls=0,failed_calls=0
cmdstat_del:calls=7506967,usec=58275084,usec_per_call=7.76,rejected_calls=400000,failed_calls=0
cmdstat_quit:calls=62,usec=191,usec_per_call=3.08,rejected_calls=0,failed_calls=0
cmdstat_hscan:calls=6,usec=114,usec_per_call=19.00,rejected_calls=0,failed_calls=0
cmdstat_FT.PROFILE:calls=2,usec=1003216,usec_per_call=501608.00,rejected_calls=0,failed_calls=0
cmdstat_select:calls=4536,usec=14916,usec_per_call=3.29,rejected_calls=0,failed_calls=0
cmdstat_hset:calls=8506964,usec=678742957,usec_per_call=79.79,rejected_calls=0,failed_calls=0
cmdstat_hlen:calls=12,usec=475,usec_per_call=39.58,rejected_calls=0,failed_calls=0
cmdstat_script|load:calls=857011,usec=558688,usec_per_call=0.65,rejected_calls=0,failed_calls=0
cmdstat_ttl:calls=17544,usec=18452,usec_per_call=1.05,rejected_calls=0,failed_calls=0
cmdstat_FT.DEL:calls=1,usec=36,usec_per_call=36.00,rejected_calls=0,failed_calls=1
cmdstat_set:calls=100000,usec=99227,usec_per_call=0.99,rejected_calls=0,failed_calls=0
cmdstat_monitor:calls=4,usec=4,usec_per_call=1.00,rejected_calls=0,failed_calls=0
cmdstat_get:calls=100083,usec=33536,usec_per_call=0.34,rejected_calls=0,failed_calls=0
cmdstat_FT.SEARCH:calls=6784,usec=420236153,usec_per_call=61945.19,rejected_calls=0,failed_calls=1
cmdstat_lrange:calls=400001,usec=5359503,usec_per_call=13.40,rejected_calls=0,failed_calls=0
cmdstat_keys:calls=5,usec=726816,usec_per_call=145363.20,rejected_calls=0,failed_calls=0
cmdstat_scan:calls=531,usec=3930340,usec_per_call=7401.77,rejected_calls=0,failed_calls=0


`redis-cli FT.PROFILE idx:string SEARCH QUERY "temp*"`
1)  1) (integer) 225799
    2) "string:296061"
    3) 1) "description"
       2) "Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum"
    4) "string:814996"
    5) 1) "description"
       2) "Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
    6) "string:661249"
    7) 1) "description"
       2) "Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum"
    8) "string:845928"
    9) 1) "description"
       2) "Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
   10) "string:290693"
   11) 1) "description"
       2) "Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
   12) "string:104680"
   13) 1) "description"
       2) "Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum"
   14) "string:356188"
   15) 1) "description"
       2) "Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
   16) "string:480494"
   17) 1) "description"
       2) "Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
   18) "string:641366"
   19) 1) "description"
       2) "Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum"
   20) "string:85477"
   21) 1) "description"
       2) "Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Excepteur sint occaecat cupidatat non proident sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet  consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
2) 1) 1) Total profile time
      2) "500"
   2) 1) Parsing time
      2) "0"
   3) 1) Pipeline creation time
      2) "0"
   4) 1) Iterators profile
      2)  1) Type
          2) UNION
          3) Query type
          4) "PREFIX - temp"
          5) Time
          6) "0"
          7) Counter
          8) (integer) 225799
          9) Child iterators
         10)  1) Type
              2) TEXT
              3) Term
              4) tempor
              5) Time
              6) "0"
              7) Counter
              8) (integer) 225799
              9) Size
             10) (integer) 900337
   5) 1) Result processors profile
      2) 1) Type
         2) Index
         3) Time
         4) "0"
         5) Counter
         6) (integer) 225799
      3) 1) Type
         2) Scorer
         3) Time
         4) "0"
         5) Counter
         6) (integer) 225799
      4) 1) Type
         2) Sorter
         3) Time
         4) "500"
         5) Counter
         6) (integer) 10
      5) 1) Type
         2) Loader
         3) Time
         4) "0"
         5) Counter
         6) (integer) 10
